Refactor imports, Add vendor/ as submodule

pradykaushik opened the pull request:
1. Refactored modules to now be prefixed with `github.com/spdf/elektron` instead of `gitlab.com/spdf/elektron`.
2. Ran `go mod init github.com/spdf/elektron` to regenerate the go.mod and go.sum files.
3. Ran `go mod vendor` to update the vendored repositories.

Note: Haven't yet changed the organization name from "spdf" to "spdfg" in the imports. We can do this once we are sure of the name.

pradykaushik committed:
ran go mod vendor on the whole project

pradykaushik committed:
refact imports to github.xxx. update go.mod + go.sum

ridv commented:
This might be a good chance to experiment with keeping  a separate repo with the dependencies. Something like elektron-vendor.

We can use git submodules to manage it. https://git-scm.com/book/en/v2/Git-Tools-Submodules

So this is how I would envision it working:

First line of defense using mod to download dependencies.
Second line, if something got moved, using the elektron-vendor repository to download the dependencies.

pradykaushik committed:
removed vendor/ folder

pradykaushik committed:
Added vendor/ submodule.

Moved dependencies that need to be vendored into separate repository.
	- https://github.com/spdfg/elektron-vendor
Added vendor/ as a submodule using the below command.
	- git submodule add https://github.com/spdfg/elektron-vendor vendor
If wanting to use vendor, run the following commands after cloning
elektron.
1. git submodule init
2. git submodule update

pradykaushik committed:
added instructions to clone vendor/ submodule.

pradykaushik commented:
@rdelval pending review.

pradykaushik committed:
updated module to spdfg. Refactored imports

ridv reviewed (approved):
:shipit:
